backend/protos/xyz/block/ftl/v1/schemaservice.proto

```diff
@@ -14,67 +14,98 @@ message GetSchemaResponse {
   repeated ftl.schema.v1.Changeset changesets = 2;
 }
 
-message PullSchemaRequest {}
-message PullSchemaResponse {
-  // ChangesetCreated is sent when a new changeset is created.
-  message ChangesetCreated {
-    ftl.schema.v1.Changeset changeset = 1;
-  }
+// ChangesetCreated is sent when a new changeset is created.
+message ChangesetCreatedNotification {
+  ftl.schema.v1.Changeset changeset = 1;
+}
 
-  // ChangesetFailed is sent when a changeset fails.
-  message ChangesetFailed {
-    string key = 1;
-    string error = 2;
-  }
+// ChangesetPreparedNotification is sent when a changeset is provisioned and ready to become active
+message ChangesetPreparedNotification {
+  ftl.schema.v1.Changeset changeset = 1;
+}
 
-  // ChangesetFailed is sent when a changeset becomes canonical.
-  message ChangesetCommitted {
-    string key = 1;
-  }
+// ChangesetFailed is sent when a changeset becomes canonical.
+message ChangesetCommittedNotification {
+  ftl.schema.v1.Changeset changeset = 1;
+}
 
-  // Deployment created is sent when a deployment is new to the listener but is not part of a changeset.
-  message DeploymentCreated {
-    // Will not be set for builtin modules.
-    // optional string key = 1;
-    // string module_name = 2;
-    // If present, the deployment is not yet canonical as it is currently part of a changeset.
-    // optional string changeset = 3;
+// ChangesetRollingBack is sent when a changeset is being rolled back without committing.
+message ChangesetRollingBackNotification {
+  string key = 1;
+  string error = 2;
+}
 
-    optional string changeset = 1;
-    optional ftl.schema.v1.Module schema = 2;
-  }
+// ChangesetFailed is sent when a changeset fails.
+message ChangesetFailedNotification {
+  string key = 1;
+  string error = 2;
+}
 
-  message DeploymentUpdated {
-    // Will not be set for builtin modules.
-    // optional string key = 1;
-    // string module_name = 2;
-    // If present, the deployment is not yet canonical as it is currently part of a changeset.
-    optional string changeset = 1;
+// ChangesetDrained is sent when a changeset has been drained.
+message ChangesetDrainedNotification {
+  string key = 1;
+}
 
-    optional ftl.schema.v1.Module schema = 2;
-  }
+// ChangesetFinalized is sent when a changeset has been finalized.
+message ChangesetFinalizedNotification {
+  string key = 1;
+}
 
-  message DeploymentRemoved {
-    // Will not be set for builtin modules.
-    optional string key = 1;
-    string module_name = 2;
+// InitialSchemaNotification is used to notify the initial schema state
+message InitialSchemaNotification {
+  optional ftl.schema.v1.Schema schema = 1;
+}
 
-    // If this is true then the module was removed as well as the deployment.
-    bool module_removed = 3;
-  }
+message DeploymentRef {
+  string deployment = 1;
+  string name = 2;
+  optional string changeset = 3;
+}
 
+message VerbRuntimeSubscriptionNotification {
```

Comment:
I think a better approach here would be to convert all runtime types into a "sum type", then have a single "RuntimeNotification" message. Otherwise we are going to have to manually keep this in sync, which will be super annoying, and we won't get sum type checking.